app/controllers/healthcheck_controller.rb

```diff
@@ -20,15 +20,35 @@ def http_status
   def green?
     # add more checks as needed:
     # database_connected? && redis_connected etc...
-    database_connected?
+    database_connected? && virus_scan_ready?
   end
 
   def database_connected?
     ActiveRecord::Base.connection.execute('SELECT 1').present?
-  rescue StandardError
+  rescue StandardError => e
+    Rails.logger.warn("Database health check failed: #{e.message}")
+
+    false
+  end
+
+  def virus_scan_ready?
+    Rails.cache.fetch('virus_scan_health_check', expires_in: 1.second) do
+      Clamby.safe?(virus_scan_test_file.to_s)
+    end
```

Comment:
☝️ we cache this test to prevents flooding of the daemon with repeated scans to this endpoint